inference/core/workflows/core_steps/models/foundation/florence_2.py

```diff
@@ -0,0 +1,274 @@
+import json
+from typing import Any, Dict, List, Literal, Optional, Type, Union
+
+import numpy as np
+import supervision as sv
+from pydantic import ConfigDict, Field
+
+from inference.core.entities.requests.inference import LMMInferenceRequest
+from inference.core.managers.base import ModelManager
+from inference.core.utils.image_utils import load_image
+from inference.core.workflows.constants import (
+    HEIGHT_KEY,
+    IMAGE_DIMENSIONS_KEY,
+    PARENT_ID_KEY,
+    ROOT_PARENT_ID_KEY,
+    WIDTH_KEY,
+)
+from inference.core.workflows.core_steps.common.entities import StepExecutionMode
+from inference.core.workflows.entities.base import (
+    Batch,
+    OutputDefinition,
+    WorkflowImageData,
+)
+from inference.core.workflows.entities.types import (
+    BATCH_OF_DETECTION_KIND,
+    BATCH_OF_DICTIONARY_KIND,
+    BATCH_OF_IMAGE_METADATA_KIND,
+    BATCH_OF_PARENT_ID_KIND,
+    LIST_OF_VALUES_KIND,
+    ROBOFLOW_MODEL_ID_KIND,
+    STRING_KIND,
+    ImageInputField,
+    StepOutputImageSelector,
+    WorkflowImageSelector,
+    WorkflowParameterSelector,
+)
+from inference.core.workflows.prototypes.block import (
+    BlockResult,
+    WorkflowBlock,
+    WorkflowBlockManifest,
+)
+
+TASKS_WITH_PROMPT = [
+    "<CAPTION_TO_PHRASE_GROUNDING>",
+    "<REFERRING_EXPRESSION_SEGMENTATION>",
+    "<REGION_TO_SEGMENTATION>",
+    "<OPEN_VOCABULARY_DETECTION>",
+    "<REGION_TO_CATEGORY>",
+    "<REGION_TO_DESCRIPTION>",
+]
+
+
+class BlockManifest(WorkflowBlockManifest):
```

Comment:
must be adjusted to be aligned with `main` - we got rid of asyncio and introduced versioning patterns. Please apply (take a look at other blocks, in doubts I can help)

Reply:
Q: What happens when LMM output does not match expectations - any error handling would be possible / error would be raised / empty output will be yielded?

Reply:
not sure what you mean by doesnt match expectation. Microsoft has already handled output validation and error handling so we can assume their preprocessor returns valid output data or empty data.